system/Database/Forge.php

```diff
@@ -1041,20 +1041,34 @@ protected function _processIndexes(string $table)
         return $sqls;
     }
 
-    protected function _processForeignKeys(string $table): string
+    /**
+     * Generates SQL to process foreign keys
+     *
+     * @param array $allowActions Allows child to override default value
+     * @param bool  $onUpdate     Allows child to avoid setting onUpdate rule (Oracle)
+     * @param bool  $shortName    Allows child to limit key name to 30 byte (Oracle)
```

Comment:
If the two parameters are used only for Oracle, it seems we don't need to have them in this parent class.

Reply:
It was a means of reusing code. The duplicate Code Detection was complaining. I suppose you could create another method instead but it would look just about like this one.

Reply:
> The duplicate Code Detection was complaining.

If it is an only issue, we can suppress the PHPCPD error.

There is no need to force premature generalizations, as they will yield poor results.
Adding unnecessary parameters to a public method is undesirable. If it is necessary for all DBs, it is fine, though.

Reply:
reverted